source/development_process/index.rst

```diff
@@ -0,0 +1,958 @@
+    # A typical ServerSpec test
+
+    require 'spec_helper'
+
+    describe '<name of the resource being tested>' do
+      # tests ...
+    end
+
+Read the `Serverspec docs`_ for more info.
+
+ChefSpec Tests
+~~~~~~~~~~~~~~
+Chefspec_ is used for `Unit Testing`_ which tests individual parts of
+a Light Weight Resource Provider (see section below for more info on what a
+LWRP is). Here's an example of a unit test from the yum chef cookbook:
+
+.. code:: ruby
+
+    require 'spec_helper'
+
+    describe 'yum::default' do
+      let(:chef_run) { ChefSpec::Runner.new.converge(described_recipe) }
+
+      it 'creates yum_globalconfig[/etc/yum.conf]' do
+        expect(chef_run).to create_yum_globalconfig('/etc/yum.conf')
+      end
+
+    end
+
+Chef Linters
+~~~~~~~~~~~~
+Chef cookbooks need to be checked just like Python code to ensure they follow
+style guidelines.
+
+
+**Rubocop_** is a Ruby static code analyzer. Out of the box it will enforce
+many of the guidelines outlined in the community `Ruby Style Guide`_. When
+`rubocop` is run, it will lint the code, display errors, and describe how to
+fix them. Rubocop can automatically fix many style errors, but this process is
+not perfect and can lead to subtle bugs. Rubocop errors should generally be
+fixed manually.
+
+Some projects may also include a `.rubocop.yml`_ file for explicitly excluding
+or including files to be analyzed by Rubocop.
+
+**`Foodcritic`_** is a linter, like Rubocop, but it enforces style guidelines
+specific to Chef cookbooks.  Foodcritic will check for conformance to `rules`_
+outlined by the Chef community, such as FC002_
+
+.. code:: bash
+
+	$ foodcritic .
+	FC002: Avoid string interpolation where not required:
+	./providers/gunicorn.rb:89
+
+.. code:: ruby
+
+	# Don't do this:
+	gunicorn_command = new_resource.virtualenv.nil? ? "gunicorn" :
+	 "#{::File.join(new_resource.virtualenv, "bin    ", "gunicorn")}"
+	# Do this instead:
+	gunicorn_command = new_resource.virtualenv.nil? ? "gunicorn" :
+	 ::File.join(new_resource.virtualenv, "bin", "gunicorn")
+
+
+
+.. _Serverspec: http://serverspec.org/
+.. _Serverspec docs: http://serverspec.org/tutorial.html
+.. _Rake: http://docs.seattlerb.org/rake/
+.. _Rubocop: http://batsov.com/rubocop/
+.. _Ruby Style Guide: https://github.com/bbatsov/ruby-style-guide
+.. _Test Kitchen: http://kitchen.ci/
+.. _Rspec: http://rspec.info/
+.. _integration testing: https://en.wikipedia.org/wiki/Integration_testing
+.. _Unit Testing: https://en.wikipedia.org/wiki/Unit_testing
+.. _Chefspec: http://sethvargo.github.io/chefspec/
+.. _.rubocop.yml: https://github.com/osuosl-cookbooks/osl-haproxy/blob/master/.rubocop.yml
+.. _Foodcritic: http://acrmp.github.io/foodcritic/
+.. _rules: http://acrmp.github.io/foodcritic/
+
+.. _FC002: http://acrmp.github.io/foodcritic/#FC002
+
+How to Write a Recipe
+---------------------
+
+- How to add dependencies
+- How to use a LWRP
+
+How to Write a Light Weight Resource Provider
```

Comment:
This is going to change a little, namely that the LWRP will no longer be defined in `resources/` and `providers/` but instead in `libraries/`. I'm preparing a short presentation about it.